bridgesync/bridgesync_test.go

```diff
@@ -771,3 +771,72 @@ func TestBridgeSync_GetLastRoot(t *testing.T) {
 		require.NotEqual(t, common.Hash{}, root.Hash)
 	})
 }
+
+func TestSubscribeToSync(t *testing.T) {
+	ctx := t.Context()
+	bridgeSyncCfg := Config{
+		DBPath:                             path.Join(t.TempDir(), "TestGetLastRoot.sqlite"),
+		BridgeAddr:                         common.HexToAddress("0x1234567890abcdef1234567890abcdef12345678"),
+		BlockFinality:                      aggkittypes.LatestBlock,
+		SyncBlockChunkSize:                 100,
+		InitialBlockNum:                    0,
+		WaitForNewBlocksPeriod:             cfgtypes.NewDuration(time.Second * 10),
+		RetryAfterErrorPeriod:              cfgtypes.NewDuration(time.Second * 5),
+		MaxRetryAttemptsAfterError:         3,
+		RequireStorageContentCompatibility: false,
+		DBQueryTimeout:                     cfgtypes.NewDuration(30 * time.Second),
+	}
+
+	mockEthClient := mocksethclient.NewEthClienter(t)
+	mockEthClient.EXPECT().CallContract(mock.Anything, mock.Anything, mock.Anything).Return(
+		common.FromHex("0x000000000000000000000000000000000000000000000000000000000000002a"), nil).Once()
```

Comment:
Double check, but I think this can be simplified. WDYT?

```suggestion
		common.FromHex("0x2a"), nil).Once()
```

Reply:
It looks like they do not give the same results. From the debugger:

```
[0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 42]
[42]
```